### PR TITLE
fix(FOROME-767): selected filters data structure

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-767

In addition:
1. Temporarily changed panels "Clear" button function.
2. Filter deletion is available only from the right refiner column (temporarily)
3. Fixed problem with Inheritance mode deletion

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by AndreyUstyumenko** [comment]
Have you thought of breaking this PR apart? I mean, 44 files, come on

File: src/pages/filter/ui/query-builder/right-refiner-column/enum-filter.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { ReactElement } from 'react'
+import Checkbox from 'react-three-state-checkbox'
+
+import { IHandleRemoveFilter } from './query-results'
+
+interface Props {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/enum-filter.tsx]: IEnumFilterProps

File: src/pages/filter/ui/query-builder/right-refiner-column/func-filter.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { ReactElement } from 'react'
+import Checkbox from 'react-three-state-checkbox'
+
+import { FuncStepTypesEnum } from '@core/enum/func-step-types-enum'
+import { TFuncArgs } from '@service-providers/common/common.interface'
+import { IHandleRemoveFilter } from './query-results'
+
+interface Props {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/func-filter.tsx]: IFuncFilterProps

File: src/pages/filter/ui/query-builder/right-refiner-column/numeric-filter.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { ReactElement } from 'react'
+import Checkbox from 'react-three-state-checkbox'
+
+import { TNumericConditionBounds } from '@service-providers/common/common.interface'
+import { getNumericExpression } from '@utils/getNumericExpression'
+
+interface Props {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/numeric-filter.tsx]: same

File: src/pages/filter/ui/query-builder/right-refiner-column/query-results.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { ReactElement } from 'react'
+import { observer } from 'mobx-react-lite'
+
+import { t } from '@i18n'
+import datasetStore from '@store/dataset'
+import filterStore from '@store/filter'
+import { SelectedFilterCard } from './selected-filter-card'
+
+export interface IHandleRemoveFilter {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/query-results.tsx]: same

File: src/pages/filter/ui/query-builder/right-refiner-column/selected-filter-card.tsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import {
+  TCondition,
+  TFuncArgs,
+  TNumericConditionBounds,
+} from '@service-providers/common/common.interface'
+import { EnumFilter } from './enum-filter'
+import { FuncFilter } from './func-filter'
+import { NumericFilter } from './numeric-filter'
+import { IHandleRemoveFilter } from './query-results'
+interface SelectedFilterCardProps {
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/selected-filter-card.tsx]: ISelectedFilterCardProps

File: src/pages/filter/ui/query-builder/right-refiner-column/selected-filter-card.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    filterId,
+    filterCondition,
+    handleRemoveFilter,
+  }: SelectedFilterCardProps): ReactElement => {
+    const [isOpen, open, close] = useToggle(true)
+
+    const filterType: string = filterCondition[0]
+    const filterName: string = filterCondition[1]
+    const filterContent: string[] = filterCondition[3]!
+    const filterExpression: TFuncArgs = filterCondition[4]!
```
Comment (AndreyUstyumenko) [inline src/pages/filter/ui/query-builder/right-refiner-column/selected-filter-card.tsx:39]: don't use non-null assertion on such things, better `filterCondition[3] || []`

File: src/pages/filter/ui/filterAttributes.store.ts [hunk 7]
```diff
@@ -135,11 +110,10 @@ export class FilterAttributesStore {
 
   private getDatasetEnumValues(group: FilterGroup): string[] {
     const { groupName } = group
+    const { conditions } = filterStore
 
     return (
-      this.datasetStore.conditions.find(
-        (element: any[]) => element[1] === groupName,
-      )?.[3] ?? []
+      conditions.find((element: any[]) => element[1] === groupName)?.[3] ?? []
```
Comment (AlMaQntr) [inline src/pages/filter/ui/filterAttributes.store.ts]: any -> TCondition

File: src/store/filter.ts [hunk 8]
```diff
@@ -1,36 +1,38 @@
...
   method!: GlbPagesNames | FilterControlOptions
   selectedGroupItem: StatListType = {}
-  dtreeSet: any = {}
-  selectedFilters: SelectedFiltersType = {}
+
   actionName?: ActionFilterEnum
   statFuncData: any = []
   filterCondition: Record<string, any> = {}
-  memorizedSelectedFilters: SelectedFiltersType | undefined = undefined
+  memorizedSelectedFilters: SelectedFiltersType | undefined | any = undefined
```
Comment (AlMaQntr) [inline src/store/filter.ts]: Please, don't use `any`

**Review by AlMaQntr — APPROVED** [review]

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**